website/notifications/events/base.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+"""Basic Event handling for events that need subscriptions"""
+
+from datetime import datetime
+
+from website.notifications import emails
+
+
+event_registry = {}
+
+
+def register(event_type):
+    """Register classes into event_register"""
+    def decorator(cls):
+        event_registry[event_type] = cls
+        return cls
+    return decorator
+
+
+class Event(object):
+    """Base event class for notification.
+
+    - abstract methods set methods that must be defined by subclasses.
```

Comment:
Yes, unfortunately now new classes need not be confined to be a subclass of an abstract class.